webui/react/src/pages/ExperimentDetails/ExperimentVisualization/ExperimentVisualizationFilters.tsx

```diff
@@ -0,0 +1,160 @@
+      [
+        ExperimentVisualizationType.HpHeatMap,
+        ExperimentVisualizationType.HpParallelCoordinates,
+        ExperimentVisualizationType.HpScatterPlots,
+      ].includes(type),
+      [
+        ExperimentVisualizationType.HpHeatMap,
+        ExperimentVisualizationType.HpParallelCoordinates,
+        ExperimentVisualizationType.HpScatterPlots,
+        ExperimentVisualizationType.LearningCurve,
+      ].includes(type),
+      [
+        ExperimentVisualizationType.HpHeatMap,
+        ExperimentVisualizationType.HpParallelCoordinates,
+        ExperimentVisualizationType.HpScatterPlots,
+      ].includes(type),
+      [ ExperimentVisualizationType.HpHeatMap ].includes(type),
+    ];
+  }, [ type ]);
+
+  const handleMaxTrialsChange = useCallback((count: SelectValue) => {
+    if (onMaxTrialsChange) onMaxTrialsChange(count as number);
+  }, [ onMaxTrialsChange ]);
+
+  const handleBatchChange = useCallback((batch: SelectValue) => {
+    if (onBatchChange) onBatchChange(batch as number);
+  }, [ onBatchChange ]);
+
+  const handleBatchMarginChange = useCallback((margin: SelectValue) => {
+    if (onBatchMarginChange) onBatchMarginChange(margin as number);
+  }, [ onBatchMarginChange ]);
+
+  const handleHParamChange = useCallback((hps: SelectValue) => {
+    if (!onHParamChange) return;
+    onHParamChange(Array.isArray(hps) && hps.length !== 0 ? hps as string[] : undefined);
+  }, [ onHParamChange ]);
+
+  const handleMetricChange = useCallback((metric: MetricName) => {
+    if (onMetricChange) onMetricChange(metric);
+  }, [ onMetricChange ]);
+
+  const handleViewChange = useCallback((view: GridListView) => {
+    if (onViewChange) onViewChange(view);
+  }, [ onViewChange ]);
+
+  return (
+    <ResponsiveFilters>
+      {showMaxTrials && (
+        <SelectFilter
+          enableSearchFilter={false}
+          label="Top Trials"
+          showSearch={false}
+          style={{ width: 70 }}
+          value={selectedMaxTrial}
+          onChange={handleMaxTrialsChange}>
+          {TOP_TRIALS_OPTIONS.map(option => <Option key={option} value={option}>{option}</Option>)}
+        </SelectFilter>
+      )}
+      {showBatches && (
+        <>
+          <SelectFilter
+            enableSearchFilter={false}
+            label="Batches Processed"
+            showSearch={false}
+            value={selectedBatch}
+            onChange={handleBatchChange}>
+            {batches.map(batch => <Option key={batch} value={batch}>{batch}</Option>)}
+          </SelectFilter>
+          <SelectFilter
+            enableSearchFilter={false}
+            label="Batch Margin"
+            showSearch={false}
+            value={selectedBatchMargin}
+            onChange={handleBatchMarginChange}>
+            {BATCH_MARGIN_OPTIONS.map(option => (
+              <Option key={option} value={option}>{option}</Option>
+            ))}
+          </SelectFilter>
+        </>
+      )}
+      {showMetrics && (
+        <MetricSelectFilter
```

Comment:
Not really related to this but most of the times I change this value I get a sequence of errors 
<img width="2434" alt="Schermata 2021-03-10 alle 15 19 21" src="https://user-images.githubusercontent.com/5413702/110643097-02712d00-81b4-11eb-8b92-5072d92d5728.png">

Reply:
and after that the page tend to crash.

Reply:
There was an uncaught error around HP parcoords data processing (unexpected undefined) that cause the crash. That should be fixed now.

Reply:
The `gd.emit` error occurs when plotly cleanup `Purge` runs against the div container that doesn't have any charts rendered. Plotly doesn't expose if a chart has rendered or not publicly, so I don't currently have a way to run the cleanup appropriately. I'll take another look at this in a separate PR for the sake of release but this warning shouldn't cause any issues.

Reply:
sounds good, and yes i see a couple of uplot errors here and there but having no side effects now.